### ISSUE TITLE
[Bug]: suboptimal output for object rest spread `({ a2, ...b2 } = c2);`

### ISSUE THREAD

**Issue by Boshen** [babel/babel#16969]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

https://github.com/babel/babel/blob/943bdfe7bc64a71389154445da3826f2ee69bbf6/packages/babel-plugin-transform-object-rest-spread/test/fixtures/object-rest/assignment-expression/input.js#L2

### Current and expected behavior

babel output:

https://github.com/babel/babel/blob/943bdfe7bc64a71389154445da3826f2ee69bbf6/packages/babel-plugin-transform-object-rest-spread/test/fixtures/object-rest/assignment-expression/output.js#L5-L10

### Environment

https://babel.dev/repl#?browsers=defaults&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=BQbwBAhgTANGB0iBGUwF8wF4wGMoEoBuIA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=&prettier=true&targets=&version=8.0.0-alpha.12&externalPlugins=%40babel%2Fplugin-transform-object-rest-spread%407.25.9&assumptions=%7B%7D

### Possible solution

tsc:

```
({ a2 } = c2, b2 = __rest(c2, ["a2"]));
```

esbuild:

```
var _a;
_a = c2, { a2 } = _a, b2 = __objRest(_a, ["a2"]);
```

### Additional context

https://github.com/babel/babel/blob/943bdfe7bc64a71389154445da3826f2ee69bbf6/packages/babel-plugin-transform-object-rest-spread/src/index.ts#L524-L561

The last line `path.replaceWithMultiple(nodes);` added the expressions as statements.

For a better output and also removing the dangling `_c;`, it should create an sequence expression instead.

**Comment by JLHwung** [comment]
Yes, we should apply the optimization technique similar to how we handle the destructuring-private proposal:

https://github.com/babel/babel/blob/943bdfe7bc64a71389154445da3826f2ee69bbf6/packages/babel-plugin-proposal-destructuring-private/src/index.ts#L163-L203
